### PR TITLE
Flux 'experimental.join()' function

### LINKED ISSUE #875: Document 'experimental.join' function
PR: https://github.com/influxdata/flux/pull/2693

### PR DESCRIPTION
**Pull request by sanderson**
Closes #875 

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/v2.0/reference/flux/stdlib/experimental/join.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+### fn
+A function with `left` and `right` arguments that maps a new output object
+using values from the `left` and `right` input objects.
+The return value must be an object.
+
+_**Data type:** Function_
+
+## Examples
+
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/join.md]: header for this example?  Join table and show difference in  field values?

File: content/v2.0/reference/flux/stdlib/experimental/join.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  |> group(columns: ["_time", "_field", "_value"], mode: "except")
+
+experimental.join(
+  left: s1,
+  right: s2,
+  fn: (left, right) => ({
+    left with
+    bar_value: left._value,
+    quz_value: right._value
+  })
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/join.md]: maybe show example output? 
what does it look like when group() groups records by columns except `_time`, `_field`, and `_value`?

Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/join.md]: That just removes `_field` from the group key. It's a bit of a clunky implementation right now. I'll circle back later and show the data stats.

File: content/v2.0/reference/flux/stdlib/experimental/join.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  v2_0_ref:
+    name: experimental.join
+    parent: Experimental
+weight: 302
+---
+
+The `experimental.join()` function joins two streams of tables on the
+[group key](/v2.0/reference/glossary/#group-key) and `_time` column.
+Columns that are not part of the group key or explicitly mapped in the join operation
+are dropped from output tables.
```
Comment (jlapacik) [inline content/v2.0/reference/flux/stdlib/experimental/join.md]: `experimental.join` will never implicitly drop columns. Whichever fields the user specifies in the `fn` argument, only those columns will be output. A user can return only a subset of the group key columns if desired.

File: content/v2.0/reference/flux/stdlib/experimental/join.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+{{% note %}}
+To join streams of tables with different fields or measuremtns, use [`group()`](/v2.0/reference/flux/stdlib/built-in/transformations/group/)
```
Comment (jlapacik) [inline content/v2.0/reference/flux/stdlib/experimental/join.md]: measurements

**Pull request merged**